### ISSUE TITLE
Cannot use exclude flag with recursive

### ISSUE THREAD

**Issue by jesseik** [Azure/azure-storage-azcopy#430]
When using AzCopy to copy content of container to new container or to new account recursively, the `--exclude` option is not honored. Instead of moving just the content of `container1` without `subdir` to location `container2/subdir` using `--exclude "subdir"`, I get structure `container2/subdir/subdir`.

### Which version of the AzCopy was used? 
AzCopy 10.1.2
### Which platform are you using? (ex: Windows, Mac, Linux)
Windows 10
### What command did you run?
```
azcopy copy "https://account.blob.core.windows.net/container1?sv=..." "https://account.blob.core.windows.net/container2/subdir?sv=..." --recursive --exclude "subdir"
```
### What problem was encountered?
Azcopy copies also subdirectory 'subdir' to destination even when set to exclude.
### How can we reproduce the problem in the simplest way?
Copy container resursively to new container or account, and try to exclude directories.
### Have you found a mitigation/solution?
No.

**Comment by zezha-msft** [comment]
Hi @jesseik, thanks for reaching out!

This looks like a bug, I've logged it for further investigation.

**Comment by JohnRusk** [comment]
I _think_ we're fixing this in 10.3.  Maybe @zezha-msft can confirm?  I'm marking it with the 10.3 milestone now, tentatively.

**Comment by zezha-msft** [comment]
@JohnRusk this depends on how the refactoring work goes.

**Comment by zezha-msft** [comment]
This has been fixed in [10.3.0](https://github.com/Azure/azure-storage-azcopy/releases/tag/v10.3.0)!

@jesseik we have split the include/exclude flags to clarify their intent. `exclude-path` flag should cover your scenario.
